### PR TITLE
Remove usage of reregistration and deprecate cli-option

### PR DESCRIPTION
**Pull request by ahitacat**
Signed-off-by: ahitacat <ahitacat@redhat.com>

### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [x] Is this PR an enhancement?

### Complete Description of Additions/Changes:

Working on deprecate --force-reregistration client option. This option will be deprecate in favor of the explicit running of `insights-client --unregister` and `insights-client --register`. This work is part of the improvement of the re-registration safeguard [RHIN-500]

Resolves: #rhbz2071082

### CODE REVIEW

**Comment by clarabez** [comment]
LGTM!

File: insights/client/config.py [hunk 1]
```diff
@@ -326,7 +326,7 @@ def _core_collect_default():
     'reregister': {
         'default': False,
         'opt': ['--force-reregister'],
-        'help': 'Forcefully reregister this machine to Red Hat. Use only as directed.',
+        'help': argparse.SUPPRESS,
```
Comment (strider) [inline insights/client/config.py:329]: As we are in deprecation mode, I will not silence that option from `argparse` now. But I would suggest adding the deprecation message, announcing the ineffectiveness of that latter and its future removal.

**Review by strider — CHANGES_REQUESTED** [review]

File: insights/client/config.py (hunk 1 above)
Comment (ahitacat) [reply insights/client/config.py]: Agree, I have added a new message

**Review by strider — APPROVED** [review]
LGTM

**Pull request merged**